src/Xrefcheck/Verify.hs

```diff
@@ -403,20 +407,37 @@ verifyReference
       VerifyResult [ExternalHttpTooManyRequests retryAfter] -> retryAfter
       _ -> Nothing
 
+    isVirtual = matchesGlobPatterns root vcVirtualFiles
+
     checkRef mAnchor referredFile = verifying $ do
+      checkReferredFileIsInsideRepo referredFile
       checkReferredFileExists referredFile
       case M.lookup referredFile repoInfo of
         Nothing -> pass  -- no support for such file, can do nothing
         Just referredFileInfo -> whenJust mAnchor $
           checkAnchor referredFile (_fiAnchors referredFileInfo)
 
+    checkReferredFileIsInsideRepo file = unless
+      (isVirtual file || noNegativeNesting (makeRelative root file))
```

Comment:
I would actually move the `isVirtual` check one level up.

Instead of checking it `checkReferredFileIsInsideRepo` and `checkReferredFileExists`, I would check it only once at the start of `checkRef`. If it points to a virtual file, then just assume the reference is valid, no further checks are necessary.